src/goto-programs/goto_convert.cpp

```diff
@@ -2061,8 +1978,8 @@ void goto_convert(
   const symbol_tablet::symbolst::const_iterator s_it=
     symbol_table.symbols.find("main");
 
-  if(s_it==symbol_table.symbols.end())
-    throw "failed to find main symbol";
+  DATA_INVARIANT(
+    s_it != symbol_table.symbols.end(), "failed to find main symbol");
```

Comment:
Is this code ever called? The hardcoded `"main"` above makes me think this must be removed as quickly as possible.